### ISSUE TITLE
Rename `Data.List.Properties.*-is-foldr` 

### ISSUE THREAD

**Issue by jamesmckinna** [agda/agda-stdlib#2516]
> I wouldn't call this `partition-is-foldr` but `partition-as-foldr` , or `partition-via-foldr`. And yes, `foldr` being the `List` eliminator, we need to be careful to not have an explosion of these. Having said that, I'm fine with `partition` as it is an important function.

But we already have
* `id-is-foldr`
* `++-is-foldr`
* `map-is-foldr`
* ...

So it does explode into a deprecation-fest if we follow your lead? v3.0?

_Originally posted by @jamesmckinna in https://github.com/agda/agda-stdlib/issues/2505#issuecomment-2506032940_
            
            
Proposal: all `*-is-fold*` lemmas be renamed, and the old names deprecated.

Issue: what alternative naming policy?

**Comment by MatthewDaggitt** [comment]
> So it does explode into a deprecation-fest if we follow your lead? v3.0?

Yes, there are lot of such names in the library. Personally my view is that the name isn't so bad that we need to go and deprecate it... but if other people are strongly in favour I'm happy to defer.
